content/posts/monitoring-with-github-actions/index.md

```diff
@@ -0,0 +1,103 @@
++++
+date = "2022-03-30T17:10:49+07:00"
+author = "ds0nt"
+description = "Using Github Actions to run some of your monitoring tasks on a schedule."
+title = "Using Github Actions for Daily Monitoring Tasks"
+categories = ["DevSecOps", "Automation", "CI"]
+tags = ["DevOps", "Github Actions", "Monitoring", "CRON", "Schedule"]
+slug = "github-actions-for-monitoring-tasks"
++++
+
+# Using Github Actions for Daily Monitoring Tasks
+
+Github actions has a CRON schedule trigger we can use run scripts on a custom schedule. It's pretty cool that Github Actions can run jobs on a CRON schedule, and we can use this ability to cover some of our dev-ops needs.
+
+For example, we used this to check if any of our domain certificates were close to their expiry. Each day, a github action  runs at 10:45, checks if our certificates are dangerously close to expiring, and posts to our `#monitoring` Slack if any of them are.
+
+
+### The Workflow
+
+This is a simple monitoring flow really; Run Check -> Alert if Problem.
+
+To do this, were going to use three Github actions.
+
+1. an action to checkout our code
+2. an action to execute a script
+3. and an action to send slack message based on our condition.
+
+```yaml
+# github/workflows/check_weather.yml`
+
+name: check_weather
```

Comment:
The previous section was talking about certificate expiry, but this section is checking weather instead. This flow is a bit awkward and confusing to me.

Maybe we can add this paragraph:
```sh
For simplification and to focus on Github Action, the following example checks and report bad weather instead of our server's certificate. In practice, we can use the example to check anything necessary.
```